keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/ec2/ClusterSpec.kt

```diff
@@ -117,16 +119,19 @@ data class ClusterSpec(
   private val _defaults: ServerGroupSpec,
   @JsonInclude(NON_EMPTY)
   val overrides: Map<String, ServerGroupSpec> = emptyMap(),
-  @JsonIgnore
-  override val deliveryArtifact: DeliveryArtifact? = null,
-  @JsonIgnore
+  // The following 3 properties are write-only as they're filled by resolvers, but not output in JSON
+  @JsonProperty(access = Access.WRITE_ONLY)
+  override val artifactType: ArtifactType? = ArtifactType.deb,
+  @JsonProperty(access = Access.WRITE_ONLY)
+  private val _artifactName: String? = null, // Custom backing field for artifactName, used by resolvers
+  @JsonProperty(access = Access.WRITE_ONLY)
   override val artifactVersion: String? = null,
```

Comment:
Artifact properties have been flattened out into discrete fields as per `ArtifactProvider` and `ArtifactReferenceProvider`.